api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java

```diff
@@ -621,6 +704,58 @@ protected String modifyTermMatch(String term) {
         .collect(Collectors.joining());
   }
 
+  protected Map<String, String> modifyTermMatchUseEndsWithTerms(String term) {
+    Map<String, String> retMap = new HashMap<>();
+
+    List<String> modifiedTerms = new ArrayList<>();
+
+    String quotedPattern = "\\\"((\\w+)\\s?)+\\\"";
+    Pattern pattern = Pattern.compile(quotedPattern);
+    Matcher matcher = pattern.matcher(term);
+    while (matcher.find()) {
+      // modifiedTerms.add("+" + matcher.group());
+      modifiedTerms.add(matcher.group());
+    }
+
+    List<String> words =
+        new ArrayList<>(Arrays.asList(term.replaceAll(quotedPattern, "").split(" ")));
+
+    List<String> endsWith =
+        words.stream()
+            .filter(word -> word.startsWith("*") && word.length() >= 4)
```

Comment:
```suggestion
            .filter(word -> word.startsWith("*") && word.length() >= 3)
```

Reply:
word length is kept to 4 because of '*' included in the word